planet/cli/orders.py

```diff
@@ -98,44 +98,96 @@ async def cancel(ctx, order_id):
     click.echo('Cancelled')
 
 
+def split_list_arg(ctx, param, value):
+    if value is None:
+        return None
+    elif value == '':
+        # note, this is specifically checking for an empty string
+        click.BadParameter('Entry cannot be an empty string.')
+
+    # split list by ',' and remove whitespace
+    entries = [i.strip() for i in value.split(',')]
+
+    # validate passed entries
+    for e in entries:
+        if not e:
+            raise click.BadParameter('Entry cannot be an empty string.')
+    return entries
+
+
+@orders.command()
+@click.pass_context
+@translate_exceptions
+@coro
+@click.argument('order_id', type=click.UUID)
+@click.option('--delay', type=int, default=5,
+              help='Time (in seconds) between polls.')
+@click.option('--max-attempts', type=int, default=5,
+              help='Maximum number of polls. Set to zero for no limit.')
+@click.option('--quiet', is_flag=True, default=False,
+              help='Disable ANSI control output.')
+@click.option('--states', type=click.STRING, callback=split_list_arg,
+              help=('One or more comma-separated order states that will end '
+                    'polling. Defaults to all completed states.'))
```

Comment:
Do you think it's possible to dependably wait on an intermediate state? I don't think it is, so perhaps we could rule those out entirely.

And I sorta feel like all completed states (success or failure) should be the same to the waiter. Allowing users to wait for failure when the process is intended to succeed 99% of the time seems not very useful.

Reply:
We had a call about this and @jreiberkyle pointed we *can* wait on an intermediate state having been reached, i.e. being at that state or a later state. So let's not rule this out. I'll resolve this conversation.

Reply:
After our conversation, I believe this could be improved to have a wait on an intermediate state return any time after that state was reached, even if polling didn't catch the state itself. I'll keep this open until I can add this improvement.